Signal/src/ViewControllers/ConversationView/MessagesViewController.m

```diff
@@ -2487,9 +2469,25 @@ - (void)updateMessageMappingRangeOptions:(MessagesRangeSizeMode)mode
 
     NSUInteger targetLength = oldLength;
     if (mode == MessagesRangeSizeMode_Truncate) {
-        // During the initial configuration of the view, we want to truncate the
-        // range size.
-        targetLength = MIN(targetLength, (NSUInteger)(kYapDatabasePageSize * kYapDatabaseMaxInitialPageCount));
+        OWSAssert(self.dynamicInteractions);
+        OWSAssert(self.page == 0);
+
+        if (self.dynamicInteractions.unreadIndicatorPosition) {
+            NSUInteger unreadIndicatorPosition
+                = (NSUInteger)[self.dynamicInteractions.unreadIndicatorPosition longValue];
+            // If there is an unread indicator, increase the initial load window
+            // to include it.
+            OWSAssert(unreadIndicatorPosition > 0);
+            OWSAssert(unreadIndicatorPosition <= kYapDatabaseRangeMaxLength);
+
+            // We'd like to include at least N seen messages,
+            // to give the user the context of where they left off the conversation.
+            const NSUInteger kPreferredSeenMessageCount = 1;
+            targetLength = unreadIndicatorPosition + kPreferredSeenMessageCount;
```

Comment:
It's really quite simple - use the unread indicator position to determine the initial range size.

Reply:
This makes sense.